specs/solver_spec.rb

```diff
@@ -0,0 +1,39 @@
+require_relative '../solver'
+
+describe Solver do
+  context 'When testing the factorial method' do
+    it 'Should return the factorial of the given positive Integer' do
+      fact = Solver.new
+      test_fact = fact.factorial(4)
+      expect(test_fact).to eq 24
+    end
```

Comment:
* Kindly add cases for where `Negative value raises an exception as an expectation` and `
Special case - when input is 0, the result should equal 1` [Refer to project requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md#:~:text=Create%20a%20method,raise%20an%20exception.)